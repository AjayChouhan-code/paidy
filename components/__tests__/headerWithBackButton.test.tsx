import React from "react";
import renderer from "react-test-renderer";
import { useNavigation } from "@react-navigation/native";
import HeaderWithBackButton from "../headerWithBackButton";
import { TouchableOpacity } from "react-native";

// Mock the useNavigation hook
jest.mock("@react-navigation/native", () => ({
  useNavigation: jest.fn(),
}));

jest.mock("@expo/vector-icons", () => {
  const React = require("react");
  return {
    Ionicons: (props) => React.createElement("Icon", props),
  };
});

describe("HeaderWithBackButton Component", () => {
  const mockGoBack = jest.fn();

  beforeEach(() => {
    (useNavigation as jest.Mock).mockReturnValue({
      goBack: mockGoBack,
    });
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it("renders correctly with a title", () => {
    const tree = renderer
      .create(<HeaderWithBackButton title="Test Title" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("calls navigation.goBack when back button is pressed", () => {
    const component = renderer.create(
      <HeaderWithBackButton title="Test Title" />
    );
    const backButton = component.root.findByType(TouchableOpacity);

    backButton.props.onPress();
    expect(mockGoBack).toHaveBeenCalled();
  });

  it("renders the back icon", () => {
    const component = renderer.create(
      <HeaderWithBackButton title="Test Title" />
    );
    const icon = component.root.findByType("Icon");

    expect(icon.props.name).toBe("arrow-back");
    expect(icon.props.size).toBe(24);
    expect(icon.props.color).toBe("black");
  });

  it("displays the correct title", () => {
    const component = renderer.create(
      <HeaderWithBackButton title="Custom Header" />
    );
    const text = component.root.findByType("Text");

    expect(text.props.children).toBe("Custom Header");
  });
});
