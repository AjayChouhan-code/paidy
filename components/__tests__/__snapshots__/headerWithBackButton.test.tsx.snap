// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderWithBackButton Component renders correctly with a title 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "padding": 10,
      "shadowRadius": 4,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "marginRight": 10,
        "opacity": 1,
      }
    }
  >
    <Icon
      color="black"
      name="arrow-back"
      size={24}
    />
  </View>
  <Text
    style={
      {
        "fontSize": 18,
        "fontWeight": "bold",
      }
    }
  >
    Test Title
  </Text>
</View>
`;
